web/src/engine/osk/src/visualKeyboard.ts

```diff
@@ -407,6 +407,22 @@ export default class VisualKeyboard extends EventEmitter<EventMap> implements Ke
       previewHost: GesturePreviewHost
     }> = {};
 
+    const clearActiveGestures = (exception?: string) => {
+      for(const identifier of Object.keys(sourceTrackingMap)) {
+        if(identifier == exception) {
+          // No need to cancel the current gesture; let the globe-key gesture complete.
+          continue;
+        }
+
+        // Any _other_ gesture, though - yeah, that should cancel out.
+        // Might be a _bit_ funky if there's an active modipress, but only momentarily.
+        const entry = sourceTrackingMap[identifier];
+
+        // Trigger cancellation of all other pending gestures - they're not valid after a keyboard-swap.
```

Comment:
The comments in this block are very specific to globe key. I think it would be better to move them outside the function and turn them into a more general function description, i.e. describe potential side-effects of using the function, etc.

i.e. `Might be a _bit_ funky` is worrying but doesn't really say what the funkiness is, etc.